eclair-core/src/main/scala/fr/acinq/eclair/PimpKamon.scala

```diff
@@ -0,0 +1,25 @@
+package fr.acinq.eclair
+
+import kamon.Kamon
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object KamonExt {
+
+  def time[T](name: String)(f: => T) = {
+    val timer = Kamon.timer(name).withoutTags().start()
+    try {
+      f
+    } finally {
+      timer.stop()
+    }
+  }
+
+  def timeFuture[T](name: String)(f: => Future[T])(implicit ec: ExecutionContext): Future[T] = {
+    val timer = Kamon.timer(name).withoutTags().start()
+    val res = f
+    res onComplete { case _ => timer.stop }
```

Comment:
Shouldn't this be `res onComplete { case Success(_) => timer.stop }`? I'd assume we don't want to mix measurements for failed calls with the successful ones.

Reply:
We need to stop the timer somehow? I don't think it matters anyway, because Kamon has first-class handling of distribution of latencies, and we would detect outliers easily.

Reply:
I think that's where we'd use tags or some equivalent. 
Usually we should split our measurements on values of a specific tag with Success, Failure1, Failure2, etc depending on our set of possible failures.